specs/core/1_custody-game.md

```diff
@@ -778,7 +796,26 @@ def after_process_final_updates(state: BeaconState) -> None:
         [record.challenger_index for record in records] + [record.responder_index for record in records]
     )
     for index, validator in enumerate(state.validators):
-        if index not in validator_indices_in_records:
-            if validator.exit_epoch != FAR_FUTURE_EPOCH and validator.withdrawable_epoch == FAR_FUTURE_EPOCH:
-                validator.withdrawable_epoch = Epoch(validator.exit_epoch + MIN_VALIDATOR_WITHDRAWABILITY_DELAY)
+        if validator.exit_epoch != FAR_FUTURE_EPOCH:
+            if (index in validator_indices_in_records 
+                    or validator.all_custody_secrets_revealed_epoch == FAR_FUTURE_EPOCH):
+                # Delay withdrawable epochs if challenge records are not empty or not all
+                # custody secrets revealed
+                validator.withdrawable_epoch = FAR_FUTURE_EPOCH
+            else:
+                # Reset withdrawable epochs if challenge records are empty
+                if validator.withdrawable_epoch == FAR_FUTURE_EPOCH:
+                    validator.withdrawable_epoch = Epoch(validator.all_custody_secrets_revealed_epoch
+                                                         + MIN_VALIDATOR_WITHDRAWABILITY_DELAY)
 ```
+
+## Additional validator initialization parameters
```

Comment:
We need to add these to the validator initialization. We also need to think how we are going to implement this change in the phase 0 to phase 1 transition; this code needs to be run for each validator when the transition happens.

Reply:
@djrtwo Do you have a good idea for where we should keep code for phase 0 -> phase 1 transition?

Reply:
I suppose we should have it in misc-beacon-changes, under a section for "fork transition scripts" or soemthing similar that makes changes to the beaconstate at transition